### ISSUE TITLE
A little question about the formula of flow loss in paper.

### ISSUE THREAD

**Issue by verlocks** [qiuyu96/CoDeF#45]
In paper, the formula of flow loss is
![image](https://github.com/qiuyu96/CoDeF/assets/45618243/9acbd35e-6529-4540-9205-f1eeca1f63b0)
and the paper introduce the $F$ as the optical flow.
However, the paper also mention 

> Corresponding points identified by flows with high confidence should be the same points in the canonical field

which fits my intuition, and I don't know why we subtract the optical flow term here?
In code I also don't find this optical flow term:
`# Optical flow loss.

            if ret.flow_loss[0] != 0:

                mk_flow_t = torch.logical_and(mk_t, flows[0].sum(dim=-1)< 3)

                loss = loss + torch.nn.functional.l1_loss(

                    ret.flow_loss[i][0][mk_flow_t], ret.flow_loss[i][1]

                    [mk_flow_t]) * self.hparams.flow_loss
`

Can you please explain why this optical flow term is in the formula? Thanks.

**Comment by JiaHeng-DLUT** [comment]
+1

**Comment by JiaHeng-DLUT** [comment]
+1

**Comment by ken-ouyang** [comment]
In the paper, the output of D is the (delta_x, delta_y) while in the code the output is (x+delta_x,y+delta_y) which includes the flow term automatically.  So both are correct.

**Comment by verlocks** [comment]
> In the paper, the output of D is the (delta_x, delta_y) while in the code the output is (x+delta_x,y+delta_y) which includes the flow term automatically. So both are correct.

Thanks for your reply, but I still don't get it. What do you mean delta_x and delta_y? In paper it says
![image](https://github.com/qiuyu96/CoDeF/assets/45618243/f51ecbf0-8fc6-4650-9432-a9f80cd1b55d)
which says the output of D is its corresponding position in canonical field. And in formula (4)
![image](https://github.com/qiuyu96/CoDeF/assets/45618243/2d93851e-61ed-43c2-8e42-75b273af2d84)
it uses the output of D directly, which should be the position in canonical field I think? So I am not sure what is delta_x and x in your reply?
